src/targets/gpu/device/include/migraphx/gpu/device/reduce.hpp

```diff
@@ -184,17 +170,8 @@ __device__ auto block_reduce(index idx, Op op, T init, ForStride fs, F f)
     }
     return y;
 }
-#endif
-template <index_int N, class Op, class T, class F>
-__device__ auto block_reduce(index idx, Op op, T init, index_int n, F f)
-{
-    auto midx = make_multi_index(idx.local, idx.nlocal());
-    // Workaround hcc, create a local array
-    auto fs = midx.id;
-    fs[0]   = n;
-    return block_reduce<N>(
-        idx, op, init, midx.for_stride(fs), [&](auto mi) __device__ { return f(mi[0]); });
-}
```

Comment:
Multi-index version is no hcc-specific. This should be kept.

Reply:
It mentioned `hcc` workround in the comment.

Reply:
Reverted change.